### PR TITLE
make shortcut logging more useful

### PR DESCRIPTION
**Pull request by nilsnolde**
log out
- total shortcuts
- total superseded edges
- avg amount of edges per shortcut for convenience

### CODE REVIEW

File: src/mjolnir/shortcutbuilder.cc [hunk 1]
```diff
@@ -337,18 +340,18 @@ bool IsEnteringEdgeOfContractedNode(GraphReader& reader, const GraphId& nodeid,
...
+                                const GraphId& start_node,
+                                const uint32_t edge_index,
+                                const uint32_t edge_count,
+                                std::unordered_map<uint32_t, uint32_t>& shortcuts) {
   // Shortcut edges have to start at a node that is not contracted - return if
   // this node can be contracted.
   EdgePairs edgepairs;
   if (CanContract(reader, tile, start_node, edgepairs)) {
-    return 0;
+    return {0UL, 0UL};
```
Comment (kevinkreiser) [inline src/mjolnir/shortcutbuilder.cc:354]: nit UL is probably 64bits wide, ie on 64 bit architectures. you could use just U I'd you wanted but if we wanted to be technically correct for all architectures you should use the exact type

File: src/mjolnir/shortcutbuilder.cc [hunk 2]
```diff
@@ -359,6 +362,7 @@ uint32_t AddShortcutEdges(GraphReader& reader,
   // Iterate through directed edges of the base node
   uint32_t shortcut = 0;
   uint32_t shortcut_count = 0;
+  int32_t total_edge_count = 0;
```
Comment (kevinkreiser) [inline src/mjolnir/shortcutbuilder.cc:365]:
```suggestion
  uint32_t total_edge_count = 0;
```

File: src/mjolnir/shortcutbuilder.cc [hunk 3]
```diff
@@ -29,6 +29,9 @@ using namespace valhalla::mjolnir;
 
 namespace {
 
+// total_shortcut_count and total_superseded_edges
+using shortcut_stats = std::pair<uint32_t, uint32_t>;
```
Comment (kevinkreiser) [inline src/mjolnir/shortcutbuilder.cc:33]: nit this typedef is only used for the return type. in the functions that return it and in the function that accumulates it its not used instead separate variables are used. we could just drop the typedef and return pair explicitly

**Review by kevinkreiser — COMMENTED** [review]
looks good just a couple nits and that typo using the signed int

File: src/mjolnir/shortcutbuilder.cc (hunk 1 above)
Comment (nilsnolde) [reply src/mjolnir/shortcutbuilder.cc:354]: I'll just do what the vast majority of valhalla is doing, passing ints:) what's ULL then on 64 bit archs?

Comment (nilsnolde) [reply src/mjolnir/shortcutbuilder.cc]: changed it to `return {uint32_t(0), uint32_t(0)};` instead

**Comment by nilsnolde** [comment]
no need for re-approval actually, just cognitive overhead:)

**Pull request merged**